torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -193,33 +213,38 @@ void LoopNestGenerator::initReduction(
       // Otherwise place it inside the last generated loop
       inner_fl->body().push_back(new_fl);
     }
+
     // Increment the inner most for loop
     inner_fl = new_fl;
   }
 
+  // Create the initialization assignment
+  const auto kir_tv = gpu_lower->lowerValue(tv);
+  const auto init_stmt = ir_builder_.create<kir::UnaryOp>(
```

Comment:
Does this mean `kir_tv` is going to have two definitions?

Reply:
Good question. No, it's going to have the definition "updated" the 2nd time - this is what I meant in the description by "a bit smaller hack than unsafe clone". We still have to come up with a complete solution.

Reply:
OK, so after the 2nd definition, the first definition is no longer tracked, right? That seems fine for now.

Reply:
Correct